for-upstream-only/nixos/profiles/nixops-hercules-ci-agent.nix

```diff
@@ -10,9 +10,10 @@
 
   users.extraUsers.hercules-ci-agent.extraGroups = [ "keys" ];
 
-  deployment.keys."agent-token.key" = {
+  deployment.keys."hercules-ci-agent-token" = {
     user = config.services.hercules-ci-agent.user;
     destDir = "/var/lib/keys/hercules-ci-agent";
```

Comment:
That apart, I still don't quite like secrets being persisted to disk in `/var/lib/keys/hercules-ci-agent` - Can you elaborate on why we don't stick to the nixops default location, `/run/keys/…` ?

Reply:
Nobody likes to persist secrets, but it is required if you want the agent to come up after a restart. Of course cloud machines may not need restarts so very often and, if restarted through NixOps, it can be made to work. We have chosen not to rely on that, in order to keep the maintenance overhead low for our users. I suggest not to change this at this time, because we have other changes planned wrt secrets in the near future.

Reply:
Well, this is a nixops-specific configuration here, so we can assume the sysadmin knows about NixOps' secret management.

Nixops suggests to let the secret-consuming systemd unit depend on `systemd.services.nixops-keys` to make sure secrets are present before starting services.
In addition, you also configured `hercules-ci-agent.service` to keep restarting indefinitely.

This is at least unintuitive compared to how nixops secret management normally works. Please at least document this deviation somewhere.

Reply:
We probably should depend on systemd service for provisioning. 

I'm not entirely sure about why does the path matter that much? Isn't it more about file permissions if you're worried about security? Those are the same either way.

The reason for not using `/run` is that it does not persist between reboots. That's why nixops supports `destDir`. 

For NixOps familiar users they can just look at Nix as single source of truth, but for all the rest we'll provide terraform as it's easier to operate and describe how secrets should be dealth with.

Reply:
Extracting secrets from a running machine is one thing, persisting secrets to disk (and leaving them there) another one - that really depends on your threat model.

I'd suggest to leave the decision about whether to persist the secret on the node or not to the user. You could suggest people to set `deployment.storeKeysOnMachine` if they want secrets to be persisted.

A much nicer option than hardcoding it into the module, and deviating from the default in an undocumented fashion - especially if plan to you target multiple deployment tools in the feature.

But well, I guess all of that can be done as a follow-up and shouldn't block this PR.

Reply:
It didn't work.
```
$ ./nixops show-physical
error: The option value `deployment.keys.hercules-ci-agent-token.path' in `/nix/store/qh4lb94846hp6sa271cn49z6qgp4k4fn-hercules-ci-agent/for-upstream-only/nixos/profiles/nixops-hercules-ci-agent.nix' is not of type `path'.
(use '--show-trace' to show detailed location information)
```

Reply:
Even when setting `path`, it seems to only be usable when [`deployment.storeKeysOnMachine`](https://github.com/NixOS/nixops/blob/master/nix/keys.nix#L217-L225) is used, otherwise `$name` is used, and the key activation script breaks halfway.